pkg/controller/operators/olm/apiservices.go

```diff
@@ -759,3 +750,54 @@ func APIServiceNameToServiceName(apiServiceName string) string {
 	// Replace all '.'s with "-"s to convert to a DNS-1035 label
 	return strings.Replace(apiServiceName, ".", "-", -1)
 }
+
+func (a *Operator) isAPIServiceAdoptable(target *v1alpha1.ClusterServiceVersion, apiService *apiregistrationv1.APIService) (adoptable bool, err error) {
+	if apiService == nil || target == nil {
+		err = errors.New("invalid input")
+		return
+	}
+
+	labels := apiService.GetLabels()
+	ownerKind := labels[ownerutil.OwnerKind]
+	ownerName := labels[ownerutil.OwnerKey]
+	ownerNamespace := labels[ownerutil.OwnerNamespaceKey]
+
+	if ownerKind == "" || ownerNamespace == "" || ownerName == "" {
+		return
+	}
+
+	if err := ownerutil.InferGroupVersionKind(target); err != nil {
+		a.logger.Warn(err.Error())
+	}	
+
+	targetKind := target.GetObjectKind().GroupVersionKind().Kind
+	if ownerKind != targetKind {
+		return
+	}
+
+	// Get the CSV that target replaces
+	replacing, replaceGetErr := a.lister.OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(target.GetNamespace()).Get(target.Spec.Replaces)
+	if replaceGetErr != nil && !k8serrors.IsNotFound(replaceGetErr) && !k8serrors.IsGone(replaceGetErr) {
+		err = replaceGetErr
+		return
+	}
+
+	// Get the current owner CSV of the APIService
+	currentOwnerCSV, ownerGetErr := a.lister.OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(ownerNamespace).Get(ownerName)
+	if ownerGetErr != nil && !k8serrors.IsNotFound(ownerGetErr) && !k8serrors.IsGone(ownerGetErr) {
+		err = ownerGetErr
+		return
+	}
+
+	owners := []ownerutil.Owner{target}
+	if replacing != nil {
+		owners = append(owners, replacing)
+	}
+	if currentOwnerCSV != nil && (
+		currentOwnerCSV.Status.Phase == v1alpha1.CSVPhaseReplacing || currentOwnerCSV.Status.Phase == v1alpha1.CSVPhaseDeleting) {
```

Comment:
I believe there's a set of phases that include these that you can check for membership